token/client/src/token.rs

```diff
@@ -2261,16 +2283,32 @@ where
             .new_decryptable_available_balance(transfer_amount, source_aes_key)
             .map_err(|_| TokenError::AccountDecryption)?;
 
-        let transfer_instruction = confidential_transfer::instruction::transfer_with_split_proofs(
-            &self.program_id,
+        let mut transfer_instruction =
+            confidential_transfer::instruction::transfer_with_split_proofs(
+                &self.program_id,
+                source_account,
+                &self.pubkey,
```

Comment:
nit: do you want consistency here with `self.get_address()`?

Reply:
Ah yes, thanks!